### DISCUSSION TITLE
Saving Site Settings on the Server...

### DISCUSSION THREAD

**Discussion by MPatriot** [oqtane/oqtane.framework#1991] [Q&A]
Everything warns us to only build modules except in a Development environment (local host), which makes sense because everything needs to be recompiled and published....

What confounds me is why would any changes to the site settings need to be done in development mode and republished?

Things like changing the alias, saving a new logo, changing the theme ect. ect?  When in development mode it saves just fine, I can see the changes in the database, like for example, if I forget to put in the www.domain.com I notice the error right away because the publishing through IIS automatically opens the server page with 'www'

Point is that for me, any attempt to save the settings and who knows what else cause the following error on the server...

System.NullReferenceException: Object reference not set to an instance of an object.
   at Oqtane.Modules.Admin.Site.Index.SaveSite() in C:\Projects\IOFlowstane\oqtane.framework\Oqtane.Client\Modules\Admin\Site\Index.razor:line 476

I must have done something wrong somewhere, isn't this the whole point in putting everything in a database?  Why is it trying to use files on my computer (my site is on a D: drive)?  Does this happen to anyone else?  I assume its something I did or something like this would have been noticed a long time ago.

Any help would be appreciated

Thanks,
- Mark

**Comment by sbwalker** [comment]
So you are saying that in your production environment when you click Save on Site Settings it throws an error? Can you please explain what version of Oqtane you are running in production, what is the hosting environment, and how did you install the framework?
